internal/pkg/aws/cloudformation/cloudformation.go

```diff
@@ -0,0 +1,233 @@
+		StackName: aws.String(stackName),
+	})
+	if err != nil {
+		if !stackDoesNotExist(err) {
+			return fmt.Errorf("delete stack %s: %w", stackName, err)
+		}
+		// Move on if stack is already deleted.
+	}
+	return nil
+}
+
+// DeleteAndWait calls Delete then blocks until the stack is deleted or until the max attempt window expires.
+func (c *CloudFormation) DeleteAndWait(stackName string) error {
+	_, err := c.client.DeleteStack(&cloudformation.DeleteStackInput{
+		StackName: aws.String(stackName),
+	})
+	if err != nil {
+		if !stackDoesNotExist(err) {
+			return fmt.Errorf("delete stack %s: %w", stackName, err)
+		}
+		return nil // If the stack is already deleted, don't wait for it.
+	}
+
+	err = c.client.WaitUntilStackDeleteCompleteWithContext(context.Background(), &cloudformation.DescribeStacksInput{
+		StackName: aws.String(stackName),
+	}, waiters...)
+	if err != nil {
+		return fmt.Errorf("wait until stack %s delete is complete: %w", stackName, err)
+	}
+	return nil
+}
+
+// Describe returns a description of an existing stack.
+// If the stack does not exist, returns ErrStackNotFound.
+func (c *CloudFormation) Describe(name string) (*StackDescription, error) {
+	out, err := c.client.DescribeStacks(&cloudformation.DescribeStacksInput{
+		StackName: aws.String(name),
+	})
+	if err != nil {
+		if stackDoesNotExist(err) {
+			return nil, &ErrStackNotFound{name: name}
+		}
+		return nil, fmt.Errorf("describe stack %s: %w", name, err)
+	}
+	if len(out.Stacks) == 0 {
+		return nil, &ErrStackNotFound{name: name}
+	}
+	descr := StackDescription(*out.Stacks[0])
+	return &descr, nil
+}
+
+// Events returns the list of stack events in **chronological** order.
+func (c *CloudFormation) Events(stackName string) ([]StackEvent, error) {
+	var nextToken *string
+	var events []StackEvent
+	for {
+		out, err := c.client.DescribeStackEvents(&cloudformation.DescribeStackEventsInput{
+			NextToken: nextToken,
+			StackName: aws.String(stackName),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("desribe stack events for stack %s: %w", stackName, err)
+		}
+		for _, event := range out.StackEvents {
+			events = append(events, StackEvent(*event))
+		}
+		nextToken = out.NextToken
+		if nextToken == nil {
+			break
+		}
+	}
+	// Reverse the events so that they're returned in chronological order.
+	// Taken from https://github.com/golang/go/wiki/SliceTricks#reversing.
+	for i := len(events)/2 - 1; i >= 0; i-- {
+		opp := len(events) - 1 - i
+		events[i], events[opp] = events[opp], events[i]
+	}
+	return events, nil
+}
+
+func (c *CloudFormation) create(stack *Stack) error {
+	return c.deployChangeSet(stack, cloudformation.ChangeSetTypeCreate)
+}
+
+func (c *CloudFormation) update(stack *Stack) error {
+	return c.deployChangeSet(stack, cloudformation.ChangeSetTypeUpdate)
+}
+
+func (c *CloudFormation) deployChangeSet(stack *Stack, changeSetType string) error {
+	cs, err := newChangeSet(c.client, stack.Name)
```

Comment:
I wonder if it makes sense to pass in the changeSetType here, rather than in cs.create

Reply:
That makes sense! 🤦‍♂ Updated :)